flink-metrics/flink-metrics-statsd/src/main/java/org/apache/flink/metrics/statsd/StatsDReporter.java

```diff
@@ -90,6 +109,45 @@ public void close() {
 
 	// ------------------------------------------------------------------------
 
+	/**
+	 * Removes leading and trailing angle brackets.
+	 */
+	private String stripBrackets(String str) {
+		return str.substring(1, str.length() - 1);
+	}
+
+	@Override
+	public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
+		if (dogstatsdMode) {
+			// memoize dogstatsd tag section: "|#tag:val,tag:val,tag:val"
+			StringBuilder statsdTagLine = new StringBuilder();
+			Map<String, String> orderedTags = new TreeMap<>(group.getAllVariables());
```

Comment:
sorry!  this was the beginning of looking at a more efficient tag table, as many entries are duplicates, but I'll take out since we're going to put the tag line in the metric object